auto sync 2022-04-20-17-35

davidvossel opened the pull request:
auto sync upstream 2022-04-20-17-35

This PR also has an update to the `hack/sync-upstream.sh` script to use a `git merge` pattern instead of `git rebase`. This gives us better control over how conflicts are handled.

nirarg commented:
/hold
wait to see the events after the fix

nirarg commented:
![image](https://user-images.githubusercontent.com/38751082/108032147-b14d9d80-703a-11eb-9fa8-3761eb2bc866.png)
Events are validated

nirarg commented:
/hold cancel

nirarg commented:
/retest

bardielle commented on pkg/actuator/actuator.go:
```diff
@@ -125,7 +125,7 @@ func (a *actuator) Create(ctx context.Context, machine *machinev1.Machine) error
 	}
 
-	err = a.kubevirtVM.Create(machineScope, userData)
+	ready, err := a.kubevirtVM.Create(machineScope, userData)
```
maybe isVMReady?

bardielle commented:
/lgtm

nirarg commented:
/retest

nunnatsa committed:
Fix the ginkgo dependencies to match v2

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

nunnatsa committed:
add printout to e2e, to check the golang version

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

nunnatsa committed:
add printout to e2e, to check the golang version

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

nunnatsa committed:
tiny change to trigger e2e test

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

nunnatsa committed:
tiny change to trigger e2e test

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

nirarg committed:
Add nirarg to the approvers list

k8s-ci-robot committed:
Merge pull request #117 from nirarg/approvers

Add nirarg to the approvers list

nunnatsa committed:
Added new CLI named clusterkubevirtadm, to manage the infra cluster

Added a new CLI tool to manage the infra cluster. For now, the tool supports two commands:

1. `clusterkubevirtadm create credentials`
   this command creates a namespace, serviceAccount, role and roleBinding in the infra cluster.
2. `clusterkubevirtadm apply credentials`
   this command creates or updatea namespace, serviceAccount, role and roleBinding in the infra cluster.
3. `clusterkubevirtadm get kubeconfig`
   this command generates a kubeconfig file to use the serviceAccount created with the first command.

The build the new tool, use:
* `make clusterkubevirtadm-linux` - for linux application
* `make clusterkubevirtadm-macos` - for macOS application
* `make clusterkubevirtadm-win` - for windows application
* `make clusterkubevirtadm-all` - for all the above + run the unit tests
* `make clusterkubevirtadm-test` - to run the unit tests

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

nunnatsa committed:
Add nunnatsa to the approvers list

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

k8s-ci-robot committed:
Merge pull request #111 from nunnatsa/clusterkubevirtadm

Added new CLI named clusterkubevirtadm, to manage the infra cluster

nunnatsa committed:
add missing go compiler to the e2e test

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

k8s-ci-robot committed:
Merge pull request #118 from nunnatsa/add-nahshon-to-owners

Add nunnatsa to the approvers list

agradouski committed:
Merge pull request #119 from nunnatsa/fix-e2e-test-go

Add missing go compiler to the e2e test

Commit:
clusterkubevirtadm-fix

k8s-ci-robot committed:
Merge pull request #120 from isaacdorfman/clusterkubevirtadm-fix

This PR is a hotfix for the broken CI, it fixes a typo in the e2e workflow

Commit:
clusterkubevirtadm-fix 2

k8s-ci-robot committed:
Merge pull request #121 from isaacdorfman/clusterkubevirtadm-fix

clusterkubevirtadm-fix 2

nirarg committed:
Fix clusterkubevirtadm-linux-amd64 calling from e2e cluster creation workflow

k8s-ci-robot committed:
Merge pull request #122 from nirarg/clusterkubevirtadm

Fix clusterkubevirtadm-linux-amd64 calling from e2e cluster creation workflow

nirarg committed:
Change clusterkubevirtadm to use --kubeconfig flag

k8s-ci-robot committed:
Merge pull request #123 from nirarg/clusterkubevirtadm

Change clusterkubevirtadm to use --kubeconfig flag

nunnatsa committed:
Restore the E2E test to not use clusterkubevirtadm

This is not working for now, and fails the E2E test lane.

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

k8s-ci-robot committed:
Merge pull request #125 from nunnatsa/fix-e2e-test-lane

Restore the E2E test to not use clusterkubevirtadm

davidvossel committed:
Update KubeVirt API mod dependency to pick up new fields

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
Set FailureReason on KubeVirtMachine when terminal state is encountered

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
Unit tests to detect terminal state

Signed-off-by: David Vossel <davidvossel@gmail.com>

agradouski committed:
refactor: move environment context out of e2e tests logic

agradouski committed:
rebase

davidvossel committed:
Switch to using rook-ceph for e2e storage class

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
rebase

nirarg committed:
Add More permissions to clusterkubevirtadm Role

The capk user need also permissions for the following resources
In order to create the tenant cluster
* virtualmachineinstances
* secrets
* services

nirarg committed:
Add More permissions to clusterkubevirtadm Role

The capk user need also permissions for the following resources
In order to create the tenant cluster
* virtualmachineinstances
* secrets
* services

k8s-ci-robot committed:
Merge pull request #128 from nirarg/clusterkubevirtadm-role

Add More permissions to clusterkubevirtadm Role

nirarg committed:
Change "./kubevirtci create-cluster" command to use clusterkubevirtadm kubeconfig

This commit changes kubevirtci to behave like the tenant cluster is created on ext infra
This is done in order to test the cluster creation with clusterkubevirtadm limited kubeconfig
This will also allow us to follow another resources would be added in the future to the tenant cluster creation
Notice: added special behavior for kubevirtci, in which the "localhost" IP in the kubeconfig is replaced
with the controlplane VM IP, in order to allow using the kubeconfig inside the cluster

k8s-ci-robot committed:
Merge pull request #114 from nunnatsa/fix-e2e-test

Fix the ginkgo dependencies to match v2

k8s-ci-robot committed:
Merge pull request #127 from davidvossel/switch-to-rook-storage-class

Switch to using rook-ceph for e2e storage class

agradouski committed:
Merge branch 'main' into e2e-env

k8s-ci-robot committed:
Merge pull request #126 from agradouski/e2e-env

refactor: move environment context out of e2e tests logic

davidvossel committed:
Add e2e test for MHC and FailureReason processing

Signed-off-by: David Vossel <davidvossel@gmail.com>

k8s-ci-robot committed:
Merge pull request #129 from nirarg/clusterkubevirtadm-kubevirtci

Change "kubevirtci create-cluster" command to use clusterkubevirtadm kubeconfig

k8s-ci-robot committed:
Merge pull request #130 from davidvossel/failure-reasons-v1

MachineHealthCheck and node remediation integration

agradouski committed:
refactor e2e tests:

- move all e2e code to e2e_test package
- consolidate logic and reduce number of files
- move required args validation to test suite initialization
- extract tenant cluster kubernetes version to env var

Signed-off-by: Alex Gradouski <agradouski@apple.com>

k8s-ci-robot committed:
Merge pull request #131 from agradouski/refactor-e2e

refactor e2e tests

davidvossel committed:
Merge remote-tracking branch 'auto-sync-script-upstream/main' into auto_sync_upstream_2022-04-20-17-35

davidvossel committed:
Update the sync script to use merge instead of rebase

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel commented:
@nunnatsa take a look at the new sync logic when you get a chance.

davidvossel commented:
the e2e tests will fail due to the transition to ginkgo v2

nunnatsa commented on hack/sync-upstream.sh:
```diff
@@ -1,95 +1,100 @@
+	git checkout  $NEW_BRANCH_NAME
+
+	DS_FILES=(OWNERS hack/sync-upstream.sh)
+	diff=$(git diff upstream/main --stat | awk '{ print $1 }')
```
low priority:

```suggestion
	diff=$(git diff upstream/main --name-only)
```

nunnatsa commented:
/lgtm

nunnatsa commented:
/test cluster-api-provider-kubevirt-e2e

nunnatsa commented:
/test cluster-api-provider-kubevirt-e2e

nunnatsa commented:
/test cluster-api-provider-kubevirt-e2e

nunnatsa commented:
/retest

nunnatsa commented:
/retest

nunnatsa commented:
/retest
